File: apk/app/src/main/java/com/example/momentra/ui/shell/personal/PersonalPulseFamily.kt
package com.example.momentra.ui.shell.personal

import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color

/** Personal Pulse family variants — Figma populated Pulse frames. */
enum class PersonalPulseFamily {
    LIFE_OPERATIONS,
    FUTURE_BUILDING,
    LIFESTYLE,
    RELATIONSHIPS,
}

fun personalPulseFamilyFor(momentTypeCode: String?): PersonalPulseFamily {
    val code = momentTypeCode?.uppercase() ?: return PersonalPulseFamily.LIFE_OPERATIONS
    return when {
        code.startsWith("LIFE_") || code == "LIFE_OPERATIONS" || code == "LIFE_RHYTHM" ->
            PersonalPulseFamily.LIFE_OPERATIONS
        code.startsWith("FUTURE_") || code == "FUTURE_BUILDING" ->
            PersonalPulseFamily.FUTURE_BUILDING
        code.startsWith("LIFESTYLE") ->
            PersonalPulseFamily.LIFESTYLE
        code.startsWith("RELATIONSHIP_") || code == "RELATIONSHIPS" ->
            PersonalPulseFamily.RELATIONSHIPS
        else -> PersonalPulseFamily.LIFE_OPERATIONS
    }
}

data class PersonalPulseFamilyTheme(
    val heroTitle: String,
    val heroSubtitleFilled: String,
    val heroSubtitleEmpty: String,
    val heroMetrics: List<String>,
    val tileLabels: List<String>,
    val nudgeTitle: String,
    val nudgeBody: String,
    val nudgeCta: String,
    val moneyTitle: String,
    val quickActions: List<String>,
    val heroStart: Color,
    val heroEnd: Color,
    val accent: Color,
)

fun PersonalPulseFamily.theme(): PersonalPulseFamilyTheme = when (this) {
    PersonalPulseFamily.LIFE_OPERATIONS -> PersonalPulseFamilyTheme(
        heroTitle = "WELLBEING SCORE",
        heroSubtitleFilled = "Your rhythm is building",
        heroSubtitleEmpty = "Awaiting first signals",
        // Figma 353:8893 hero chips
        heroMetrics = listOf("Pressure", "Recovery", "Discipline", "Attention"),
        // Figma metric tiles (2x2)
        tileLabels = listOf("Pressure", "Recovery", "Discipline", "Attention"),
        nudgeTitle = "Protect Recovery",
        nudgeBody = "Add a recovery block before your next busy stretch.",
        nudgeCta = "Log Recovery Now",
        moneyTitle = "MONEY SNAPSHOT",
        quickActions = listOf("Recovery", "Attention", "Mood", "Money", "Adjust"),
        heroStart = Color(0xFF7C5CFC),
        heroEnd = Color(0xFFA78BFA),
        accent = Color(0xFF7C5CFC),
    )
    PersonalPulseFamily.FUTURE_BUILDING -> PersonalPulseFamilyTheme(
        heroTitle = "FUTURE SCORE",
        heroSubtitleFilled = "Your trajectory is strong",
        heroSubtitleEmpty = "Awaiting first signals",
        heroMetrics = listOf("Vision", "Growth", "Momentum", "Discipline"),
        tileLabels = listOf("Vision", "Growth", "Momentum", "Discipline"),
        nudgeTitle = "Accelerate Growth",
        nudgeBody = "Log a milestone to keep momentum compounding.",
        nudgeCta = "Log Milestone",
        moneyTitle = "INVESTMENT SNAPSHOT",
        quickActions = listOf("Milestone", "Opportunity", "Pivot", "Progress", "Learning"),
        // SCREEN_STALE fix (G8): align with MomentThemes emerald, not purple Pulse leftover.
        heroStart = Color(0xFF10B981),
        heroEnd = Color(0xFF34D399),
        accent = Color(0xFF10B981),
    )
    PersonalPulseFamily.LIFESTYLE -> PersonalPulseFamilyTheme(
        // Figma `505:12365` — VITALITY INDEX axes: Joy / Fulfillment / Vitality / Exploration
        heroTitle = "VITALITY INDEX",
        heroSubtitleFilled = "Network stability · Flourishing",
        heroSubtitleEmpty = "Awaiting first signals",
        heroMetrics = listOf("Joy", "Fulfillment", "Vitality", "Exploration"),
        tileLabels = listOf("Joy", "Fulfillment", "Vitality", "Exploration"),
        nudgeTitle = "Protect a ritual",
        nudgeBody = "Log one experience to protect your lifestyle rhythm.",
        nudgeCta = "Log Experience",
        moneyTitle = "LIFESTYLE SPEND",
        quickActions = listOf("Experience", "Wellbeing", "Discovery", "Create", "Adjust"),
        heroStart = Color(0xFF0EA5A4),
        heroEnd = Color(0xFF7C5CFC),
        accent = Color(0xFF7C5CFC),
    )
    PersonalPulseFamily.RELATIONSHIPS -> PersonalPulseFamilyTheme(
        // Figma `505:11793` — BOND INDEX axes include Trust / Care (+ Support / Presence)
        heroTitle = "BOND INDEX",
        heroSubtitleFilled = "Stable and deepening",
        heroSubtitleEmpty = "Awaiting first signals",
        heroMetrics = listOf("Trust", "Care", "Support", "Presence"),
        tileLabels = listOf("Trust", "Care", "Support", "Presence"),
        nudgeTitle = "Protect Connection",
        nudgeBody = "Log a connection before the next busy stretch.",
        nudgeCta = "Log Connection",
        moneyTitle = "SHARED SPEND",
        quickActions = listOf("Connection", "Shared", "Investment", "Support", "Adjust"),
        heroStart = Color(0xFFE91E63),
        heroEnd = Color(0xFFA78BFA),
        accent = Color(0xFFE12A9E),
    )
}
fun PersonalPulseFamilyTheme.heroBrush(): Brush =
    Brush.horizontalGradient(listOf(heroStart, heroEnd))

File: momentra/momentra/Shell/Personal/Shared/PersonalPulseFamily.swift
import SwiftUI

enum PersonalPulseFamily: Equatable {
    case lifeOperations
    case futureBuilding
    case lifestyle
    case relationships

    static func forTypeCode(_ momentTypeCode: String?) -> PersonalPulseFamily {
        guard let raw = momentTypeCode?.uppercased() else { return .lifeOperations }
        if raw.hasPrefix("LIFE_") || raw == "LIFE_OPERATIONS" || raw == "LIFE_RHYTHM" {
            return .lifeOperations
        }
        if raw.hasPrefix("FUTURE_") || raw == "FUTURE_BUILDING" {
            return .futureBuilding
        }
        if raw.hasPrefix("LIFESTYLE") {
            return .lifestyle
        }
        if raw.hasPrefix("RELATIONSHIP_") || raw == "RELATIONSHIPS" {
            return .relationships
        }
        return .lifeOperations
    }
}

struct PersonalPulseFamilyTheme {
    let heroTitle: String
    let heroSubtitleFilled: String
    let heroSubtitleEmpty: String
    let heroMetrics: [String]
    let tileLabels: [String]
    let nudgeTitle: String
    let nudgeBody: String
    let nudgeCta: String
    let moneyTitle: String
    let quickActions: [String]
    let heroStart: Color
    let heroEnd: Color
    let accent: Color
}

extension PersonalPulseFamily {
    var theme: PersonalPulseFamilyTheme {
        switch self {
        case .lifeOperations:
            return PersonalPulseFamilyTheme(
                heroTitle: "WELLBEING SCORE",
                heroSubtitleFilled: "Your rhythm is building",
                heroSubtitleEmpty: "Awaiting first signals",
                heroMetrics: ["Pressure", "Recovery", "Discipline", "Attention"],
                tileLabels: ["Pressure", "Recovery", "Discipline", "Attention"],
                nudgeTitle: "Protect Recovery",
                nudgeBody: "Add a recovery block before your next busy stretch.",
                nudgeCta: "Log Recovery Now",
                moneyTitle: "MONEY SNAPSHOT",
                quickActions: ["Recovery", "Attention", "Mood", "Money", "Adjust"],
                heroStart: Color(hex: "#7C5CFC"),
                heroEnd: Color(hex: "#A78BFA"),
                accent: Color(hex: "#7C5CFC")
            )
        case .futureBuilding:
            // SCREEN_STALE fix (G8): align with MomentThemes / matrix emerald `#10B981` / `#34D399`.
            return PersonalPulseFamilyTheme(
                heroTitle: "FUTURE SCORE",
                heroSubtitleFilled: "Your trajectory is strong",
                heroSubtitleEmpty: "Awaiting first signals",
                heroMetrics: ["Vision", "Growth", "Momentum", "Discipline"],
                tileLabels: ["Vision", "Growth", "Momentum", "Discipline"],
                nudgeTitle: "Accelerate Growth",
                nudgeBody: "Log a milestone to keep momentum compounding.",
                nudgeCta: "Log Milestone",
                moneyTitle: "INVESTMENT SNAPSHOT",
                quickActions: ["Milestone", "Opportunity", "Pivot", "Progress", "Learning"],
                heroStart: Color(hex: "#10B981"),
                heroEnd: Color(hex: "#34D399"),
                accent: Color(hex: "#10B981")
            )
        case .lifestyle:
            return PersonalPulseFamilyTheme(
                heroTitle: "VITALITY INDEX",
                heroSubtitleFilled: "Network stability · Flourishing",
                heroSubtitleEmpty: "Awaiting first signals",
                heroMetrics: ["Joy", "Fulfillment", "Vitality", "Exploration"],
                tileLabels: ["Joy", "Fulfillment", "Vitality", "Exploration"],
                nudgeTitle: "Protect a ritual",
                nudgeBody: "Log one experience to protect your lifestyle rhythm.",
                nudgeCta: "Log Experience",
                moneyTitle: "LIFESTYLE SPEND",
                quickActions: ["Experience", "Wellbeing", "Discovery", "Create", "Adjust"],
                heroStart: Color(hex: "#0EA5A4"),
                heroEnd: Color(hex: "#7C5CFC"),
                accent: Color(hex: "#7C5CFC")
            )
        case .relationships:
            return PersonalPulseFamilyTheme(
                heroTitle: "BOND INDEX",
                heroSubtitleFilled: "Stable and deepening",
                heroSubtitleEmpty: "Awaiting first signals",
                heroMetrics: ["Trust", "Care", "Support", "Presence"],
                tileLabels: ["Trust", "Care", "Support", "Presence"],
                nudgeTitle: "Protect Connection",
                nudgeBody: "Log a connection before the next busy stretch.",
                nudgeCta: "Log Connection",
                moneyTitle: "SHARED SPEND",
                quickActions: ["Connection", "Shared", "Investment", "Support", "Adjust"],
                heroStart: Color(hex: "#E91E63"),
                heroEnd: Color(hex: "#A78BFA"),
                accent: Color(hex: "#E12A9E")
            )
        }
    }
}

func pulseQuickActionSymbol(_ label: String) -> String {
    switch label.lowercased() {
    case "recovery": return "waveform.path.ecg"
    case "attention": return "scope"
    case "mood", "reflection", "reflect": return "face.smiling"
    case "money": return "wallet.pass"
    case "adjust", "inbox": return "slider.horizontal.3"
    case "milestone", "progress", "growth", "learning", "opportunity", "post": return "chart.line.uptrend.xyaxis"
    case "chat", "check-in", "plan", "support", "presence": return "person.2"
    default: return "bolt.fill"
    }
}
